File: rust/kqt-android/app/src/main/java/plus/meow/kqt/MainActivity.java
package plus.meow.kqt;

import android.app.NativeActivity;
import android.content.Intent;
import android.os.Bundle;

public class MainActivity extends NativeActivity {

    static {
        System.loadLibrary("kqt_android");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);

        notifyOnNewIntent();
    }

    private native void notifyOnNewIntent();
}

File: rust/kqt-android/app/src/main/kotlin/plus/meow/kqt/MainActivity.kt
package plus.meow.kqt

import android.app.NativeActivity
import android.content.Intent
import android.os.Bundle

class MainActivity : NativeActivity() {

    companion object {
        init {
            System.loadLibrary("kqt_android")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)

        notifyOnNewIntent()
    }

    private external fun notifyOnNewIntent()
}
